python/cudf/cudf/core/dataframe.py

```diff
@@ -3929,7 +3939,7 @@ def scatter_by_map(self, map_index, map_size=None):
 
             # Append empty dataframes if map_size > len(tables)
             for i in range(map_size - len(tables)):
-                tables.append(self.iloc[[]])
+                tables.append(cudf.DataFrame(columns=self.columns))
```

Comment:
Instead of returning an empty typed Index this will return a RangeIndex now, is that intended?

Reply:
Good point -  I was missing a check for empty-dataframe index types in `test_dataframe_scatter_by_map`, so I didn't notice this mistake.  I added a simple fix with test coverage.